Application/Published/Upgrade-AdminApp/Deploy-EdFiOdsAdminApp.ps1

```diff
@@ -0,0 +1,220 @@
+.PARAMETER AdminAppVersion
+	Admin app version to be deployed. Defaults to 2.2.0.
+.PARAMETER Edition
+	Edition (Test, Release) of the Azure Deploy scripts to deploy.  Defaults to Release.
+.EXAMPLE
+	.\Deploy-EdFiOds.ps1 -ResourceGroupName "Ed-Fi-Ods-Resourcegroup" -AdminAppName "AdminApp-Latest" -AppInsightLocation "South Central US" -ProductionApiUrl "https://edfiodsapiwebsite-production-yuw8iui32.azurewebsites.net"
+    Deploys the provided version of the AdminApp to the South Central US Azure region with the default instance name
+#>
+Param(
+
+	[string] 
+	[Parameter(Mandatory=$true)] 
+	$ResourceGroupName,
+
+	[string] 
+	[Parameter(Mandatory=$true)] 
+	$AdminAppName,
+
+	[ValidateSet("East US", "South Central US")]
+	[string] 
+	[Parameter(Mandatory=$true)]	
+	$AppInsightLocation,
+	
+	[string]
+	[Parameter(Mandatory=$true)]
+	$ProductionApiUrl,
+
+	[string]
+	[Parameter(Mandatory=$true)]
+	$SQLServerHostname,
+
+	[string]
+	[Parameter(Mandatory=$true)]
+	$SQLServerUserName,
+
+	[SecureString]
+	[Parameter(Mandatory=$true)]
+	$SQLServerPassword,
+
+	[string]
+	[Parameter(Mandatory=$false)] 
+	$EncryptionKey,
+
+	[string] 
+	[Parameter(Mandatory=$false)] 
+	$TemplateFileDirectory = '$PSScriptRoot',
+
+	[string]
+	[Parameter(Mandatory=$false)]
+	$AdminAppVersion = '2.2.0',
+	
+	[string]
+	[ValidateSet('Release','Test')]
+	$Edition = 'Release'	
+)
+
+Import-Module $PSScriptRoot\..\Dependencies.psm1 -Force -DisableNameChecking
+Import-Module $PSScriptRoot\..\EdFiOdsDeploy.psm1 -Force -DisableNameChecking
+
+Use-Module "AzureRM" "4.3.1"
+Use-Module "AzureRM.profile" "3.3.1"
+
+# Powershell doesn't support using $PSScriptRoot in parameter defaults,
+# so using that value as a marker to update with the real PSScriptRoot value
+if ($TemplateFileDirectory -eq '$PSScriptRoot')
+{
+	$TemplateFileDirectory = $PSScriptRoot;
+}
+
+$CacheTimeOut = "10"
+
+$AdminAppTemplateFile = "$TemplateFileDirectory\OdsAdminApp.Upgrade.json"
+$AdminAppTemplateParametersFile = "$TemplateFileDirectory\OdsAdminApp.parameters.Upgrade.json"
+
+function Deploy-AdminApp()
+{
+	if(-not $EncryptionKey)
+	{
+		$aes = [System.Security.Cryptography.Aes]::Create()
+		$aes.KeySize = 256
+		$aes.GenerateKey()
+		$EncryptionKey = [System.Convert]::ToBase64String($aes.Key)
+	}
+	
+	$deployParameters = New-Object -TypeName Hashtable
+	$deployParameters.Add("version", $AdminAppVersion)
+	$deployParameters.Add("edition", $Edition)
+	$deployParameters.Add("appInsightsLocation", $AppInsightLocation)
+	$deployParameters.Add("odsInstanceName", $ResourceGroupName)
+	$deployParameters.Add("sqlServerAdminLogin", $SQLServerUserName)
+	$deployParameters.Add("sqlServerAdminPassword", $SQLServerPassword)
+	$deployParameters.Add("productionApiUrl", $ProductionApiUrl)	
+	$deployParameters.Add("metadataCacheTimeOut", $CacheTimeOut)
+	$deployParameters.Add("adminAppNameToDeploy", $AdminAppName)
+	$deployParameters.Add("encryptionKey", $EncryptionKey)
```

Comment:
You don't need to change anything... just commenting for future reference. Typically I see this written in a simpler format:
```
	$deployParameters = @{
	    version = $AdminAppVersion
	    edition = $Edition
	    appInsightsLocation = $AppInsightLocation
            ... etc
        }
```